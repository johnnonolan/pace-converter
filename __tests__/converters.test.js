const {
  minkm2kmh,
  kmh2minkm,
  minkm2minmi,
  minmi2minkm,
  minkm2mih,
  mih2minkm,
  kmh2minmi,
  minmi2kmh,
  kmh2mih,
  mih2kmh,
  mih2minmi,
  minmi2mih
} = require('../src/converters');

describe('converters', () => {
  const scenarios = [
    {
      minkm: 4.0,
      kmh: 15.0,
      minmi: 6.437,
      mih: 9.321
    },
    {
      minkm: 4.661,
      kmh: 12.873,
      minmi: 7.5,
      mih: 8.0
    },
    {
      minkm: 5.0,
      kmh: 12.0,
      minmi: 8.046,
      mih: 7.457
    },
    {
      minkm: 6.0,
      kmh: 10.0,
      minmi: 9.656,
      mih: 6.214
    },
    {
      minkm: 7.0,
      kmh: 8.571,
      minmi: 11.265,
      mih: 5.326
    },
    {
      minkm: 8.0,
      kmh: 7.5,
      minmi: 12.873,
      mih: 4.661
    }
  ];

  describe('minkm2kmh', () => {
    scenarios.forEach((scenario) => {
      test(`expect ${scenario.minkm} minkm to be ${scenario.kmh} kmh`, () => {
        const minkm = scenario.minkm;
        const kmh = minkm2kmh(minkm);
        expect(Math.abs(kmh - scenario.kmh)).toBeLessThan(0.0011);
        expect(kmh).toMatchSnapshot();
      });
    });
  });

  describe('kmh2minkm', () => {
    scenarios.forEach((scenario) => {
      test(`expect ${scenario.kmh} kmh to be ${scenario.minkm} minkm`, () => {
        const kmh = scenario.kmh;
        const minkm = kmh2minkm(kmh);
        expect(Math.abs(minkm - scenario.minkm)).toBeLessThan(0.0011);
        expect(minkm).toMatchSnapshot();
      });
    });
  });

  describe('minkm2minmi', () => {
    scenarios.forEach((scenario) => {
      test(`expect ${scenario.minkm} minkm to be ${scenario.minmi} minmi`, () => {
        const minkm = scenario.minkm;
        const minmi = minkm2minmi(minkm);
        expect(Math.abs(minmi - scenario.minmi)).toBeLessThan(0.0011);
        expect(minmi).toMatchSnapshot();
      });
    });
  });

  describe('minmi2minkm', () => {
    scenarios.forEach((scenario) => {
      test(`expect ${scenario.minmi} minmi to be ${scenario.minkm} minkm`, () => {
        const minmi = scenario.minmi;
        const minkm = minmi2minkm(minmi);
        expect(Math.abs(minkm - scenario.minkm)).toBeLessThan(0.0011);
        expect(minkm).toMatchSnapshot();
      });
    });
  });

  describe('minkm2mih', () => {
    scenarios.forEach((scenario) => {
      test(`expect ${scenario.minkm} minkm to be ${scenario.mih} mih`, () => {
        const minkm = scenario.minkm;
        const mih = minkm2mih(minkm);
        expect(Math.abs(mih - scenario.mih)).toBeLessThan(0.0011);
        expect(mih).toMatchSnapshot();
      });
    });
  });

  describe('mih2minkm', () => {
    scenarios.forEach((scenario) => {
      test(`expect ${scenario.mih} mih to be ${scenario.minkm} minkm`, () => {
        const mih = scenario.mih;
        const minkm = mih2minkm(mih);
        expect(Math.abs(minkm - scenario.minkm)).toBeLessThan(0.0011);
        expect(minkm).toMatchSnapshot();
      });
    });
  });

  describe('kmh2minmi', () => {
    scenarios.forEach((scenario) => {
      test(`expect ${scenario.kmh} kmh to be ${scenario.minmi} minmi`, () => {
        const kmh = scenario.kmh;
        const minmi = kmh2minmi(kmh);
        expect(Math.abs(minmi - scenario.minmi)).toBeLessThan(0.0011);
        expect(minmi).toMatchSnapshot();
      });
    });
  });

  describe('minmi2kmh', () => {
    scenarios.forEach((scenario) => {
      test(`expect ${scenario.minmi} minmi to be ${scenario.kmh} kmh`, () => {
        const minmi = scenario.minmi;
        const kmh = minmi2kmh(minmi);
        expect(Math.abs(kmh - scenario.kmh)).toBeLessThan(0.0011);
        expect(kmh).toMatchSnapshot();
      });
    });
  });

  describe('kmh2mih', () => {
    scenarios.forEach((scenario) => {
      test(`expect ${scenario.kmh} kmh to be ${scenario.mih} mih`, () => {
        const kmh = scenario.kmh;
        const mih = kmh2mih(kmh);
        expect(Math.abs(mih - scenario.mih)).toBeLessThan(0.0011);
        expect(mih).toMatchSnapshot();
      });
    });
  });

  describe('mih2kmh', () => {
    scenarios.forEach((scenario) => {
      test(`expect ${scenario.mih} mih to be ${scenario.kmh} kmh`, () => {
        const mih = scenario.mih;
        const kmh = mih2kmh(mih);
        expect(Math.abs(kmh - scenario.kmh)).toBeLessThan(0.0011);
        expect(kmh).toMatchSnapshot();
      });
    });
  });

  describe('mih2minmi', () => {
    scenarios.forEach((scenario) => {
      test(`expect ${scenario.mih} mih to be ${scenario.minmi} minmi`, () => {
        const mih = scenario.mih;
        const minmi = mih2minmi(mih);
        expect(Math.abs(minmi - scenario.minmi)).toBeLessThan(0.0011);
        expect(minmi).toMatchSnapshot();
      });
    });
  });

  describe('minmi2mih', () => {
    scenarios.forEach((scenario) => {
      test(`expect ${scenario.minmi} minmi to be ${scenario.mih} mih`, () => {
        const minmi = scenario.minmi;
        const mih = minmi2mih(minmi);
        expect(Math.abs(mih - scenario.mih)).toBeLessThan(0.0011);
        expect(mih).toMatchSnapshot();
      });
    });
  });
});
