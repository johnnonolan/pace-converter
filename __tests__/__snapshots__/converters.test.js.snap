// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converters kmh2mih expect 7.5 kmh to be 4.661 mih 1`] = `4.661`;

exports[`converters kmh2mih expect 8.571 kmh to be 5.326 mih 1`] = `5.326`;

exports[`converters kmh2mih expect 10 kmh to be 6.214 mih 1`] = `6.214`;

exports[`converters kmh2mih expect 12 kmh to be 7.457 mih 1`] = `7.457`;

exports[`converters kmh2mih expect 12.873 kmh to be 8 mih 1`] = `7.999`;

exports[`converters kmh2mih expect 15 kmh to be 9.321 mih 1`] = `9.321`;

exports[`converters kmh2minkm expect 7.5 kmh to be 8 minkm 1`] = `8`;

exports[`converters kmh2minkm expect 8.571 kmh to be 7 minkm 1`] = `7`;

exports[`converters kmh2minkm expect 10 kmh to be 6 minkm 1`] = `6`;

exports[`converters kmh2minkm expect 12 kmh to be 5 minkm 1`] = `5`;

exports[`converters kmh2minkm expect 12.873 kmh to be 4.661 minkm 1`] = `4.661`;

exports[`converters kmh2minkm expect 15 kmh to be 4 minkm 1`] = `4`;

exports[`converters kmh2minmi expect 7.5 kmh to be 12.873 minmi 1`] = `12.874`;

exports[`converters kmh2minmi expect 8.571 kmh to be 11.265 minmi 1`] = `11.265`;

exports[`converters kmh2minmi expect 10 kmh to be 9.656 minmi 1`] = `9.656`;

exports[`converters kmh2minmi expect 12 kmh to be 8.046 minmi 1`] = `8.046`;

exports[`converters kmh2minmi expect 12.873 kmh to be 7.5 minmi 1`] = `7.501`;

exports[`converters kmh2minmi expect 15 kmh to be 6.437 minmi 1`] = `6.437`;

exports[`converters mih2kmh expect 4.661 mih to be 7.5 kmh 1`] = `7.501`;

exports[`converters mih2kmh expect 5.326 mih to be 8.571 kmh 1`] = `8.571`;

exports[`converters mih2kmh expect 6.214 mih to be 10 kmh 1`] = `10`;

exports[`converters mih2kmh expect 7.457 mih to be 12 kmh 1`] = `12`;

exports[`converters mih2kmh expect 8 mih to be 12.873 kmh 1`] = `12.874`;

exports[`converters mih2kmh expect 9.321 mih to be 15 kmh 1`] = `15`;

exports[`converters mih2minkm expect 4.661 mih to be 8 minkm 1`] = `7.999`;

exports[`converters mih2minkm expect 5.326 mih to be 7 minkm 1`] = `7`;

exports[`converters mih2minkm expect 6.214 mih to be 6 minkm 1`] = `6`;

exports[`converters mih2minkm expect 7.457 mih to be 5 minkm 1`] = `5`;

exports[`converters mih2minkm expect 8 mih to be 4.661 minkm 1`] = `4.661`;

exports[`converters mih2minkm expect 9.321 mih to be 4 minkm 1`] = `4`;

exports[`converters mih2minmi expect 4.661 mih to be 12.873 minmi 1`] = `12.873`;

exports[`converters mih2minmi expect 5.326 mih to be 11.265 minmi 1`] = `11.265`;

exports[`converters mih2minmi expect 6.214 mih to be 9.656 minmi 1`] = `9.656`;

exports[`converters mih2minmi expect 7.457 mih to be 8.046 minmi 1`] = `8.046`;

exports[`converters mih2minmi expect 8 mih to be 7.5 minmi 1`] = `7.5`;

exports[`converters mih2minmi expect 9.321 mih to be 6.437 minmi 1`] = `6.437`;

exports[`converters minkm2kmh expect 4 minkm to be 15 kmh 1`] = `15`;

exports[`converters minkm2kmh expect 4.661 minkm to be 12.873 kmh 1`] = `12.873`;

exports[`converters minkm2kmh expect 5 minkm to be 12 kmh 1`] = `12`;

exports[`converters minkm2kmh expect 6 minkm to be 10 kmh 1`] = `10`;

exports[`converters minkm2kmh expect 7 minkm to be 8.571 kmh 1`] = `8.571`;

exports[`converters minkm2kmh expect 8 minkm to be 7.5 kmh 1`] = `7.5`;

exports[`converters minkm2mih expect 4 minkm to be 9.321 mih 1`] = `9.321`;

exports[`converters minkm2mih expect 4.661 minkm to be 8 mih 1`] = `7.999`;

exports[`converters minkm2mih expect 5 minkm to be 7.457 mih 1`] = `7.457`;

exports[`converters minkm2mih expect 6 minkm to be 6.214 mih 1`] = `6.214`;

exports[`converters minkm2mih expect 7 minkm to be 5.326 mih 1`] = `5.326`;

exports[`converters minkm2mih expect 8 minkm to be 4.661 mih 1`] = `4.661`;

exports[`converters minkm2minmi expect 4 minkm to be 6.437 minmi 1`] = `6.437`;

exports[`converters minkm2minmi expect 4.661 minkm to be 7.5 minmi 1`] = `7.501`;

exports[`converters minkm2minmi expect 5 minkm to be 8.046 minmi 1`] = `8.046`;

exports[`converters minkm2minmi expect 6 minkm to be 9.656 minmi 1`] = `9.656`;

exports[`converters minkm2minmi expect 7 minkm to be 11.265 minmi 1`] = `11.265`;

exports[`converters minkm2minmi expect 8 minkm to be 12.873 minmi 1`] = `12.874`;

exports[`converters minmi2kmh expect 6.437 minmi to be 15 kmh 1`] = `15`;

exports[`converters minmi2kmh expect 7.5 minmi to be 12.873 kmh 1`] = `12.874`;

exports[`converters minmi2kmh expect 8.046 minmi to be 12 kmh 1`] = `12.001`;

exports[`converters minmi2kmh expect 9.656 minmi to be 10 kmh 1`] = `10`;

exports[`converters minmi2kmh expect 11.265 minmi to be 8.571 kmh 1`] = `8.571`;

exports[`converters minmi2kmh expect 12.873 minmi to be 7.5 kmh 1`] = `7.501`;

exports[`converters minmi2mih expect 6.437 minmi to be 9.321 mih 1`] = `9.321`;

exports[`converters minmi2mih expect 7.5 minmi to be 8 mih 1`] = `8`;

exports[`converters minmi2mih expect 8.046 minmi to be 7.457 mih 1`] = `7.457`;

exports[`converters minmi2mih expect 9.656 minmi to be 6.214 mih 1`] = `6.214`;

exports[`converters minmi2mih expect 11.265 minmi to be 5.326 mih 1`] = `5.326`;

exports[`converters minmi2mih expect 12.873 minmi to be 4.661 mih 1`] = `4.661`;

exports[`converters minmi2minkm expect 6.437 minmi to be 4 minkm 1`] = `4`;

exports[`converters minmi2minkm expect 7.5 minmi to be 4.661 minkm 1`] = `4.661`;

exports[`converters minmi2minkm expect 8.046 minmi to be 5 minkm 1`] = `5`;

exports[`converters minmi2minkm expect 9.656 minmi to be 6 minkm 1`] = `6`;

exports[`converters minmi2minkm expect 11.265 minmi to be 7 minkm 1`] = `7`;

exports[`converters minmi2minkm expect 12.873 minmi to be 8 minkm 1`] = `7.999`;
